plugin/src/main/java/org/wildfly/plugin/provision/PackageServerMojo.java

```diff
@@ -305,9 +342,45 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             getLog().debug(String.format("Skipping " + getGoal() + " of %s:%s", project.getGroupId(), project.getArtifactId()));
             return;
         }
+        deploymentResolution = DeploymentResolution.getInstance(project);
         super.execute();
     }
 
+    private void deploy(Path deploymentContent, String targetName) throws MojoDeploymentException {
+        if (Files.exists(deploymentContent)) {
+            Path standaloneDeploymentDir = Path.of(provisioningDir, "standalone", "deployments");
+            if (!standaloneDeploymentDir.isAbsolute()) {
+                standaloneDeploymentDir = Path.of(project.getBuild().getDirectory()).resolve(standaloneDeploymentDir);
+            }
+            try {
+                Path deploymentTarget = standaloneDeploymentDir.resolve(targetName);
+                getLog().info("Copy deployment " + deploymentContent + " to " + deploymentTarget);
+                Files.copy(deploymentContent, deploymentTarget, StandardCopyOption.REPLACE_EXISTING);
+            } catch (IOException e) {
+                throw new MojoDeploymentException("Could not copy deployment in provisioned server", e);
+            }
+        }
+    }
+
+    private Map<String, Path> getDeployments() throws MavenUniverseException, MojoExecutionException {
+        if (deploymentPaths == null) {
+            deploymentPaths = new LinkedHashMap<>();
+            for (DeploymentConfiguration c : deployments) {
+                Path p;
+                File f = deploymentResolution.getFile(c.getGroupId(), c.getArtifactId(), c.getClassifier(),
+                        c.getType());
+                if (f == null) {
+                    throw new MojoExecutionException("Deployment not found " + c.getGroupId() + ":" + c.getArtifactId()
+                            + (c.getClassifier() == null ? "" : ":" + c.getClassifier()) + ":" + c.getType());
```

Comment:
Minor, but probably useful. We could do this logic in the `DeploymentConfiguration.toString()`.

Reply:
Done.